torchok/tasks/detection.py

```diff
@@ -68,15 +74,18 @@ def forward_with_gt(self, batch: Dict[str, Union[torch.Tensor, int]]) -> Dict[st
         """Forward with ground truth labels."""
         input_data = batch.get('image')
         features = self.backbone.forward_features(input_data)[-self.num_scales:]
-        with torch.cuda.amp.autocast(dtype=torch.float32):
-            neck_out = self.neck(features)
+        neck_out = self.neck(features)
         prediction = self.bbox_head(neck_out)
         output = self.bbox_head.format_dict(prediction)
         output['image_shape'] = input_data.shape[-2:]
 
-        if 'bboxes' in batch:
-            output['gt_bboxes'] = batch.get('bboxes')
-            output['gt_labels'] = batch.get('label')
+        if 'bboxes' in batch.keys():
+            if 'num_bboxes' in batch.keys():
+                output['gt_bboxes'] = [bbox[:n, :4] for bbox, n in zip(batch['bboxes'], batch['num_bboxes'])]
+                output['gt_labels'] = [bbox[:n, 4].long() for bbox, n in zip(batch['bboxes'], batch['num_bboxes'])]
+            else:
+                output['gt_bboxes'] = batch.get('bboxes')[:, :, :4]
```

Comment:
Is it COCO dataset has't `num_bboxes`? If yes why it has't?

Reply:
There may be another dataset